### ISSUE TITLE
Update to latest AWS CDK version (2.151.0)

### ISSUE THREAD

**Issue by hshepherd** [aws-quickstart/cdk-eks-blueprints#1058]
### Describe the feature

Would it be possible to get a new release of blueprints that has `peerDependencies` on aws-cdk and aws-cdk-lib v2.151.0?

e.g.

NatGatewayProvider: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.NatGatewayProvider.html
aws-cdk release: https://github.com/aws/aws-cdk/releases/tag/v2.151.0

### Use Case

We have a requirement to be able to use the newly exported `NatGatewayProvider` in `aws-cdk-lib` version `2.151.0`. 
Since we are using cdk-eks-blueprints in the same project -- and blueprints is locked to 2.147.3 -- we cannot yet use this class. 

### Proposed Solution

```json
// package.json
"peerDependencies": {
    "aws-cdk-lib": "2.150.0",
    "aws-cdk": "2.150.0"
}
```

### Acknowledgements

- [X] I may be able to implement this feature request

### CDK version used

2.147.3

### EKS Blueprints Version

1.15.1

### Node.js Version

18.20.4

### Environment details (OS name and version, etc.)

OSX 13.4

**Comment by shapirov103** [comment]
You should be able to upgrade your project to the latest CDK by overriding the version of the CDK in your package.json and running with `--force` flag, similar to what we use in our patterns repo [here](https://github.com/aws-samples/cdk-eks-blueprints-patterns/blob/3b873e04e07522834c922341dec64be58a18054b/package.json#L51-L52).

**Comment by hshepherd** [comment]
OK I will give this a try if this is a supported pattern
Thanks for the response!

**Comment by shapirov103** [comment]
This was addressed in #1087.
